### DISCUSSION TITLE
USD Values

### DISCUSSION THREAD

**Discussion by alfredpedroza** [davetapley/helium-tax#27] [Q&A] [answered]
For the last few days, since about 8/12/2021, I noticed the USD value does not fluctuate/update as much as before according to the actual market value. Is this typical and will sort itself out (API/blockchain issues, etc) or did something happen that needs to be corrected with the Helium Tax tool? Thanks for a great tool!!

**Comment by davetapley** [comment, marked as answer]
Well spotted!
So the USD value is calculated using [the Oracle Price](https://docs.helium.com/blockchain/oracles), and that was intentionally designed to be somewhat resilient to fluctuations. Salient parts from [the docs](https://docs.helium.com/blockchain/oracles/#calculating-the-hnt-price):

> nine oracles (this number may change in the future) periodically submit HNT/$USD prices. Once the blockchain has enough new price data, it will calculate a new HNT/$USD price that will remain valid until enough new, valid oracle inputs are submitted, triggering a new price revision.


> A valid price submission is anything that was submitted within the last 25 hours but is older than 1 hour. This enables the blockchain to calculate a trailing 24 hour median while also having a buffer against outlier price inputs in the most recent 60 minutes.

I chose to use the Oracle Price because it was:
1. Reasonably ease to implement (since I was already interacting with the Helium API), and:
2. Has the benefit of being tolerant by design to individual exchanges being broken/unavailable.

As far as it pertains to the IRS and tax code, [the IRS says](https://www.irs.gov/irb/2014-16_IRB#NOT-2014-21):

> Q–5: How is the fair market value of virtual currency determined?
> 
> A–5: For U.S. tax purposes, transactions using virtual currency must be reported in U.S. dollars. Therefore, taxpayers will be required to determine the fair market value of virtual currency in U.S. dollars as of the date of payment or receipt. If a virtual currency is listed on an exchange and the exchange rate is established by market supply and demand, the fair market value of the virtual currency is determined by converting the virtual currency into U.S. dollars (or into another real currency which in turn can be converted into U.S. dollars) at the exchange rate, in a reasonable manner that is consistently applied.

I would hope that the Oracle Price meets the criteria of "a reasonable manner that is consistently applied",
but I am not a tax person so 🤷🏻‍♂️

**Reply by alfredpedroza** [reply]
I appreciate the explanation and links! Glad to know this is a non-issue and by design. Thanks again!
